### PR TITLE
Reduce Dependencies

### PR DESCRIPTION
**Pull request by realmayus**
This PR aims to clean up a few dependencies listed in https://github.com/corona-school/project-user/issues/449.

* Bumped node version to 18.x to support native `fetch`
* Implemented custom mailjet solution to avoid having to ship the SDK
* Migrated the S3 SDK to v3 which is modular, resulting in a smaller dependency size

### CODE REVIEW

File: common/mails/mailjet.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { mailjetSmtp } from "./config";
-import * as mailjetAPI from "node-mailjet";
 import { getLogger } from "log4js";
+import * as mailjet from "./mailjetTypes";
 
 const logger = getLogger();
+const auth = btoa(`${mailjetSmtp.auth.user}:${mailjetSmtp.auth.pass}`);
```
Comment (Jonasdoubleyou) [inline common/mails/mailjet.ts:6]: `Buffer.from(...).toString('base64')` ...

**Comment by Jonasdoubleyou** [comment]
Um why should I review this when this doesn't build? There's another `btoa` ....

**Comment by realmayus** [comment]
Oops, forgot to run the code 😅

**Review by Jonasdoubleyou — DISMISSED** [review]
Have you tested the mail sending? We currently send out a few hundred mails each day, so if that fails it gets us in some trouble ...

**Comment by realmayus** [comment]
Just tested the email sending again, and it works both on my machine and on Heroku. I tested both directly sent emails and notifications, so it *should* work.

File: web/controllers/expertController/index.ts [hunk 2]
```diff
@@ -265,7 +265,7 @@ async function putExpert(wixId: string, student: Student, info: ApiPutExpert): P
 async function GetExpertiseTagEntities(tagNames: string[]): Promise<ExpertiseTag[]> {
     const entityManager = getManager();
 
-    const tags: ExpertiseTag[] = await entityManager.find(ExpertiseTag, { where: tagNames.map(t => ({ name: t }))});
+    const tags: ExpertiseTag[] = await entityManager.find(ExpertiseTag, { where: { name: In(tagNames)}});
```
Comment (realmayus) [inline web/controllers/expertController/index.ts:268]: This suddenly wouldn't compile anymore.... Maybe it has to do with the updated Node version? 
The In() is equivalent to the tagNames.map() function in this case, right?

Comment (realmayus) [reply web/controllers/expertController/index.ts:268]: Concrete error:
```
Error:(268, 11) TS2322: Type '{ name: string; }[]' is not assignable to type 'ExpertiseTag[]'.
  Type '{ name: string; }' is missing the following properties from type 'ExpertiseTag': id, expertData
 ```

Comment (Jonasdoubleyou) [reply web/controllers/expertController/index.ts:268]: no repro. Probably something wrong with your build setup

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**